pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -13,19 +13,47 @@
 from pyomo.core.expr.current import ExpressionReplacementVisitor
 
 from pyomo.common.modeling import unique_component_name
+from pyomo.common.deprecation import deprecated
 from pyomo.opt import SolverFactory
+import logging
 
+logger = logging.getLogger('pyomo.contrib.sensitivity_toolbox')
 
-
+@deprecated("The sipopt function has been deprecated. Use the sensitivity_calculation() "
+            "function with method='sipopt' to access this functionality.",
+            logger='pyomo.contrib.sensitivity_toolbox',
+            version='TBD')
 def sipopt(instance, paramSubList, perturbList,
-           cloneModel=True, streamSoln=False, keepfiles=False):
-    """This function accepts a Pyomo ConcreteModel, a list of parameters, along
-    with their corresponding perterbation list. The model is then converted
-    into the design structure required to call sipopt to get an approximation
-    perturbed solution with updated bounds on the decision variable. 
+           cloneModel=True, streamSoln=False, keepfiles=False):    
+    m = sensitivity_calculation('sipopt', instance, paramSubList, perturbList,
+         cloneModel, streamSoln, keepfiles, optarg=None)
+
+    return m
+
+@deprecated("The kaug function has been deprecated. Use the sensitivity_calculation() "
+            "function with method='kaug' to access this functionality.", 
+            logger='pyomo.contrib.sensitivity_toolbox',
+            version='TBD')
+def kaug(instance, paramSubList, perturbList,
+         cloneModel=True, streamSoln=False, keepfiles=False, optarg=None):
+    m = sensitivity_calculation('kaug', instance, paramSubList, perturbList,
+         cloneModel, streamSoln, keepfiles, optarg)
+
+    return m
+
+def sensitivity_calculation(method, instance, paramSubList, perturbList,
+         cloneModel=True, streamSoln=False, keepfiles=False, optarg=None):
```

Comment:
I recognize that `tee` is not the most obvious option, but I think that you should consider using it here for consistency with other solvers / interfaces.  If we decide to start supporting an option name other than `tee` we should do it for all of pyomo.  My feeling is that having each package do its own thing is actually more confusing for the user (as they now have to look up/remember each interface).

Similarly, you might consider making `optarg` follow the "standard" pattern of `solver_options`.

Reply:
I changed them to follow "standards": 
- `streamSoln` -> `tee`
- `optarg` -> `solver_options`